primo/data_parser/well_data.py

```diff
@@ -0,0 +1,1115 @@
+            # age_estimation(self)
+            raise_exception(
+                "Age estimation feature is not supported currently.",
+                NotImplementedError,
+            )
+
+    def _check_depth_availability(self):
+        """
+        Check if the depth of the well is available or not. If not, then
+        this method fills the missing information.
+        """
+        # Identify wells for which depth is not specified
+        col_name = self._col_names.depth
+        flag, empty_cells = self.has_incomplete_data(col_name)
+        if not flag:
+            # Depth is provided for all wells
+            return
+
+        # Flag the wells for which the depth is estimated.
+        self.flag_wells(empty_cells, "depth_flag")
+
+        if self.config.missing_depth == "fill":
+            LOGGER.warning(
+                f"Assigning the depth of the well as {self.config.fill_depth} "
+                f"ft, if it is missing. To change this number, pass fill_depth "
+                f"argument while instantiating the WellData object."
+            )
+            self.fill_incomplete_data(col_name, self.config.fill_depth)
+
+        elif self.config.missing_depth == "remove":
+            self.drop_incomplete_data(col_name, "depth")
+
+        elif self.config.missing_depth == "estimate":
+            LOGGER.warning(
+                "Estimating the depth of a well, if it is missing, from its "
+                "nearest neighbors."
+            )
+            # depth_estimation(self)
+            raise_exception(
+                "Depth estimation feature is not supported currently.",
+                NotImplementedError,
+            )
+
+    def _filter_production_volume(self):
+        """
+        Removes wells whose lifelong production volume is less
+        than a specified value.
+        """
+        wcn = self._col_names
+        if wcn.life_gas_production is None and wcn.life_oil_production is None:
+            LOGGER.warning(
+                "Unable to filter wells based on production volume because both lifelong gas "
+                "production [in Mcf] and lifelong oil production [in bbl] are missing in the "
+                "input data. Please provide at least one of the columns via attributes "
+                "life_gas_production and life_oil_production in the WellDataColumnNames object."
+            )
+            return
+
+        self._removed_rows["production_volume"] = []
+        if wcn.life_gas_production in self:
+            self.fill_incomplete_data(
+                col_name=wcn.life_gas_production,
+                value=self.config.fill_life_gas_production,
+                flag_col_name="life_gas_production_flag",
+            )
+            # Remove wells if their production volume is greater than the threshold
+            production_volume = self.config.threshold_gas_production
+            remove_rows = self.data[
+                self.data[wcn.life_gas_production] > production_volume
```

Comment:
In this section, where the data is being filtered, the criterion for dropping the data is compared strictly (i.e., > threshold) against the threshold. Later (line 821) in classifying the wells' depth, the comparison against the threshold is not strict (>=). I think consistency is good for the comparisons, but I also understand that the contexts (i.e., one is for filtering the data and the other is for classifying the data) are different.

Reply:
Yeah, I agree. I used a strict inequality because the earlier case study used a strict inequality. I changed it to `>=` in the latest commit.